ce/api/util.py

```diff
@@ -114,4 +122,18 @@ def search_for_unique_ids(sesh, ensemble_name='ce', model='', emission='',
     if timescale:
         query = query.filter(mm.TimeSet.time_resolution == timescale)
 
+    if cell_method:
+        if not validate_cell_method(cell_method):
+            raise Exception('Unsupported cell_method: {}'.format(cell_method))
+
+        cell_methods = sesh.query(mm.DataFileVariable.variable_cell_methods)\
+                        .distinct(mm.DataFileVariable.variable_cell_methods).all()
+        pattern = {
+            'standard_deviation': r'time: [a-z]* time: standard_deviation over days',
+            'mean': r'time: [a-z]*( time: mean .*)?'
+        }[cell_method]
+
+        matching_cell_methods = [r[0] for r in cell_methods if re.match(pattern, r[0])]
+        query = query.filter(mm.DataFileVariable.variable_cell_methods.in_(matching_cell_methods))
```

Comment:
FWIW, you can use regular expressions directly in PostgreSQL with the `~` operator. In SQLAlchemy, this looks something like this:

```
pattern = ...
session.query(Object).filter(Object.column.op('regexp')(pattern))
```

This would cut your queries in half and make the code more clear and direct.

Reply:
I had tried that (and a bunch of alternatives) but I could never get it to work.  From what I had found it wasn't supported, hence why I created a separate query.  If I'm missing something I'd be happy to filter that way instead but I haven't been able to get it to work.